### DISCUSSION TITLE
SQL string from prepared statement

### DISCUSSION THREAD

**Discussion by maxwell8888** [rusqlite/rusqlite#1161] [Questions] [answered]
Say I have a prepared statement like below
```rust
let mut stmt = conn.prepare("INSERT INTO test (foo, var) VALUES (?, ?)")?;
stmt.execute(params![1i32, 2i32])?;
```
Is there a way to get the prepared statement as a string with the parameters bound? i.e. `"INSERT INTO test (foo, var) VALUES (1, 2)"`. This would be for logging, debugging, etc. The only solution I can think of is to construct the string myself with something like `format!("INSERT INTO test (foo, var) VALUES ({}, {})", 1, 2)` but this isn't a good equivalent, for example I loose the string escaping that gets done. I have a feeling that this is not possible because from a quick google it doesn't seem like sqlite provides this functionality, but I'm a relative beginner and have only ever really used the CLI. Could this be an additional feature that rusqlite could provide? Hopefully it is clear what I am trying to achieve and if I am going about this the wrong way, I would appreciate being pointed in the right direction.

**Comment by gwenn** [comment, marked as answer]
https://github.com/rusqlite/rusqlite/blob/46def6d67e27ab5d77cdf3895455f75d7b05ce91/src/statement.rs#L1413-L1415
